### PR TITLE
gapic: support EnableJwtWithScope for gRPC GAPIC clients

### PR DESCRIPTION
**Pull request by shinfan**
- Update the GAPIC clients to allow scope for self-signed JWT
- The self-signed JWT is now the default authentication method for service account instead of OAuth 2.0
- Self-signed JWT AIP: https://google.aip.dev/auth/4111
- Related PRs:
  -  https://github.com/googleapis/google-api-go-client/pull/1075
  -  https://github.com/golang/oauth2/pull/504
  - https://github.com/aip-dev/google.aip.dev/pull/764

### CODE REVIEW

**Comment by shinfan** [comment]
cc @codyoss @silvolu @bshaffer

**Comment by noahdietz** [comment]
googleapis/google-api-go-client#1075 needs to be released and we might need to update the version of google.golang.org/api used in the `showcase` integration test [submodule](https://github.com/googleapis/gapic-generator-go/blob/master/showcase/go.mod#L9). The integration tests do not compile atm because of this.

LGTM otherwise

**Comment by shinfan** [comment]
> LGTM otherwise

Thanks Noah. I will fix the dependency issue once it is released.

**Comment by codyoss** [comment]
Released: https://github.com/googleapis/google-api-go-client/releases/tag/v0.49.0

**Comment by shinfan** [comment]
@noahdietz PTAL

File: internal/gengapic/log [hunk 1]
```diff
@@ -0,0 +1,351 @@
+--- FAIL: TestClientOpt (0.00s)
```
Comment (noahdietz) [inline internal/gengapic/log:1]: Please remove this.

Comment (shinfan) [reply internal/gengapic/log]: Oops. Done.

**Review by noahdietz — APPROVED** [review]

**Pull request merged**